tools/parse/src/main/scala/org/allenai/nlpstack/parse/poly/fsm/EmbeddedClassifier.scala

```diff
@@ -96,30 +85,58 @@ class DTCostFunctionTrainer(
       trainingVectorSource.groupVectorIteratorsByTask
 
     var progressCounter = 1
-    (taskTrainingVectors map {
-      case (task, trainingVectorIter) => {
-        val trainingVectors = trainingVectorIter.toIterable
-        println(s"Task ${progressCounter} of ${trainingVectorSource.tasks.size}" +
-          s"(${task.filenameFriendlyName}) has ${trainingVectors.size} training vectors")
+    var taskClassifierSeq: Seq[(ClassificationTask, File)] = Seq() // TODO: do we need a var here?
```

Comment:
I don't think so. `taskTrainingVectors.map` should do.

Why do you put `ClassificationTask` into this `Seq` in the first place? It's never used, from what I can tell. It looks like a `Seq[File]` will do.